src/odemis/acq/test/move_test.py

```diff
@@ -0,0 +1,188 @@
+                                     atol=ATOL_STAGE)
+
+    def test_imaging_from_loading(self):
+        """
+        Test moving the sample stage to imaging position while it's in loading position
+        """
+        stage = self.stage
+        # Get the stage to loading position
+        f = stage.moveAbs(stage.getMetadata()[model.MD_FAV_POS_DEACTIVE])
+        f.result()
+        f = cryoLoadSample(IMAGING)
+        f.result()
+        test.assert_pos_almost_equal(stage.position.value, stage.getMetadata()[model.MD_FAV_POS_ACTIVE],
+                                     atol=ATOL_STAGE)
+
+    def test_imaging_from_tilting(self):
+        """
+        Test moving the sample stage to imaging position while it's in tilting position
+        """
+        stage = self.stage
+        focus = self.focus
+        # Get the stage to tilting position and park focus
+        f = stage.moveAbs(stage.getMetadata()[model.MD_FAV_POS_ACTIVE])
+        f.result()
+        f = focus.moveAbs(focus.getMetadata()[model.MD_FAV_POS_DEACTIVE])
+        f.result()
+        f = stage.moveAbs({'rx': 0.003, 'rz': 0.003})
+        f.result()
+        f = cryoTiltSample(rx=0, rz=0)
+        f.result()
+        test.assert_pos_almost_equal(stage.position.value, {'rx': 0, 'rz': 0}, match_all=False,
+                                     atol=ATOL_STAGE)
+
+    def test_tilting_from_imaging(self):
+        """
+        Test tilting the sample stage while the stage is in Imaging position
+        """
+        stage = self.stage
+        focus = self.focus
+        # Get the stage to imaging position and park focus
+        f = stage.moveAbs(stage.getMetadata()[model.MD_FAV_POS_ACTIVE])
+        f.result()
+        f = focus.moveAbs(focus.getMetadata()[model.MD_FAV_POS_DEACTIVE])
+        f.result()
+        f = cryoTiltSample(rx=0.003, rz=0.003)
+        f.result()
+        test.assert_pos_almost_equal(stage.position.value, {'rx': 0.003, 'rz': 0.003}, match_all=False,
+                                     atol=ATOL_STAGE)
+
+    def test_tilting_from_loading(self):
+        """
+        Test it's not possible to do tilting movement while the stage is in Loading position
+        """
+        stage = self.stage
+        f = stage.moveAbs(stage.getMetadata()[model.MD_FAV_POS_DEACTIVE])
+        f.result()
+        with self.assertRaises(ValueError):
+            f = cryoTiltSample(rx=0.003, rz=0.003)
+            f.result()
+
+    def test_cancel_loading(self):
+        """
+        Test cryoLoadSample movement cancellation is handled correctly
+        """
+        stage = self.stage
+        f = stage.moveAbs(stage.getMetadata()[model.MD_FAV_POS_ACTIVE])
+        f.result()
+        f = cryoLoadSample(LOADING)
+        time.sleep(2)
+        cancelled = f.cancel()
+        self.assertTrue(cancelled)
+        with self.assertRaises(AssertionError):
+            test.assert_pos_almost_equal(stage.position.value, stage.getMetadata()[model.MD_FAV_POS_DEACTIVE],
```

Comment:
Just a note, and it's fine to leave you're code like it is: this feels like we're missing a `test.assert_pos_not_equal()` function.

Reply:
Indeed, we've recently added `assert_array_not_equal()`, and a `assert_pos_not_almost_equal()` could be handy. In particular, instead of this implementation, if the keys are different, it should (still) fail.

Reply:
Cool, I will include `assert_pos_not_almost_equal` to the `test` module.